src/da/da-helper.js

```diff
@@ -0,0 +1,560 @@
+      simplifiedAssetMapping.set(url, `/${fullShadowPath}/${filename}`);
+    } catch (error) {
+      // If URL parsing fails, assume it's already a relative path
+      const filename = url.split('/').pop();
+      simplifiedAssetMapping.set(url, `/${fullShadowPath}/${filename}`);
+    }
+  });
+  
+  return simplifiedAssetMapping;
+}
+
+/**
+ * Download assets for a page to the shadow folder
+ * @param {Array<string>} matchingHrefs - Array of matching asset URLs
+ * @param {string} fullShadowPath - The full shadow folder path
+ * @param {string} downloadFolder - Base download folder
+ * @param {number} maxRetries - Maximum retries for download
+ * @param {number} retryDelay - Delay between retries
+ * @param {Object} dependencies - Dependencies for testing (optional)
+ * @return {Promise<Array>} Download results
+ */
+async function downloadPageAssets(matchingHrefs, fullShadowPath, downloadFolder, maxRetries, retryDelay, dependencies = defaultDependencies) {
+  const { chalk: chalkDep } = dependencies;
+  
+  const simplifiedAssetMapping = createSimplifiedAssetMapping(matchingHrefs, fullShadowPath, dependencies);
+  
+  console.log(chalkDep.blue(`Downloading ${matchingHrefs.length} assets for this page...`));
+  const downloadResults = await dependencies.downloadAssets(simplifiedAssetMapping, downloadFolder, maxRetries, retryDelay);
+  
+  // Count successful downloads
+  const successfulDownloads = downloadResults.filter(result => result.status === 'fulfilled').length;
+  const failedDownloads = downloadResults.filter(result => result.status === 'rejected').length;
+  
+  console.log(chalkDep.green(`Successfully downloaded ${successfulDownloads} assets`));
+  if (failedDownloads > 0) {
+    console.log(chalkDep.red(`Failed to download ${failedDownloads} assets`));
+  }
+  
+  return downloadResults;
+}
+
+/**
+ * Upload assets for a page to DA
+ * @param {string} shadowFolderPath - Path to the shadow folder containing assets
+ * @param {string} daLocation - DA location URL
+ * @param {string} token - Authentication token
+ * @param {Object} uploadOptions - Upload options
+ * @param {string} downloadFolder - Base download folder for relative path calculation
+ * @param {number} pageIndex - Page index for logging
+ * @param {Object} dependencies - Dependencies for testing (optional)
+ * @return {Promise<void>}
+ */
+async function uploadPageAssets(shadowFolderPath, daLocation, token, uploadOptions, downloadFolder, pageIndex, dependencies = defaultDependencies) {
+  const { chalk: chalkDep, uploadFolder: uploadFolderDep } = dependencies;
+  
+  console.log(chalkDep.yellow(`Uploading assets for page ${pageIndex} to ${shadowFolderPath}...`));
+  try {
+    await uploadFolderDep(shadowFolderPath, daLocation, token, {
+      ...uploadOptions,
+      fileExtensions: ['.jpg', '.jpeg', '.png', '.gif', '.svg', '.webp', '.pdf'],
+      excludePatterns: ['node_modules', '.git'],
+      verbose: false, // Less verbose for individual page uploads
```

Comment:
```suggestion
```
I don't think we need an check for valid extension or exclusions - if it is present in the download folder, upload it.